source/simulator/checkpoint_restart.cc

```diff
@@ -388,6 +366,41 @@ namespace aspect
 
     }
 
+    // Now rename the snapshots to put the new one in place of the old one.
```

Comment:
Can you add an MPI_Barrier before this line, please? I am a little nervous that not all processors might be done writing the ``restart.mesh`` when processor 0 gets here. This wasn't a problem with the old code, as the ``mv`` happened before the next snapshot only.